File: wagerproof-ios-native/Wagerproof/Features/Search/Components/SearchResultRow.swift
import SwiftUI
import WagerproofDesign

/// Standard iOS list row used by every `SearchView` result section. One row
/// type for all four result kinds (games, agents, outliers, scores) so the
/// search list reads as a uniform surface — only the icon color and the
/// secondary label change between kinds.
///
/// Layout (matches Honeydew's settings-row + system Spotlight rhythm):
///   - 28pt leading icon in a tinted rounded square
///   - Two-line text stack (primary 17pt, secondary 13pt secondary)
///   - Trailing chevron (system standard) — search results always push or
///     present, so the chevron is non-negotiable
///
/// The accessory is rendered as a `Image(systemName: "chevron.right")`
/// rather than relying on `NavigationLink`'s built-in chevron because the
/// SearchView's result rows fire imperative actions (switch tab + open
/// sheet) instead of pushing a child view on the local stack.
struct SearchResultRow: View {
    /// SF Symbol name. `trophy.fill` for games, `brain.head.profile` for
    /// agents, `bell.badge.fill` for outliers, `sportscourt.fill` for scores.
    let icon: String
    /// Tint applied to the leading icon's background. Each kind picks its
    /// own brand-aligned tint so a glance at the column tells the user what
    /// surface a row belongs to.
    let tint: Color
    let primary: String
    let secondary: String?
    /// Optional trailing micro-label (e.g. live score "21-7" or agent net
    /// units "+12.4u"). Rendered between the secondary text and the
    /// chevron, smaller and monospaced.
    let trailingDetail: String?
    let onTap: () -> Void

    init(
        icon: String,
        tint: Color,
        primary: String,
        secondary: String? = nil,
        trailingDetail: String? = nil,
        onTap: @escaping () -> Void
    ) {
        self.icon = icon
        self.tint = tint
        self.primary = primary
        self.secondary = secondary
        self.trailingDetail = trailingDetail
        self.onTap = onTap
    }

    var body: some View {
        Button(action: onTap) {
            HStack(spacing: 12) {
                iconBadge
                VStack(alignment: .leading, spacing: 2) {
                    Text(primary)
                        .font(.system(size: 16, weight: .semibold))
                        .foregroundStyle(Color.appTextPrimary)
                        .lineLimit(1)
                    if let secondary, !secondary.isEmpty {
                        Text(secondary)
                            .font(.system(size: 13))
                            .foregroundStyle(Color.appTextSecondary)
                            .lineLimit(1)
                    }
                }
                Spacer(minLength: 8)
                if let trailingDetail, !trailingDetail.isEmpty {
                    Text(trailingDetail)
                        .font(.system(size: 13, weight: .medium, design: .monospaced))
                        .foregroundStyle(Color.appTextSecondary)
                        .lineLimit(1)
                }
                Image(systemName: "chevron.right")
                    .font(.system(size: 13, weight: .semibold))
                    .foregroundStyle(Color.appTextMuted)
            }
            .padding(.vertical, 8)
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
    }

    /// Tinted rounded-square icon container. 36pt outer square, 18pt symbol.
    /// Same shape language as the Settings rows so search results feel
    /// native to the system UI.
    private var iconBadge: some View {
        ZStack {
            RoundedRectangle(cornerRadius: 8, style: .continuous)
                .fill(tint.opacity(0.16))
            Image(systemName: icon)
                .font(.system(size: 16, weight: .semibold))
                .foregroundStyle(tint)
        }
        .frame(width: 36, height: 36)
    }
}

#Preview("Search result rows") {
    List {
        SearchResultRow(
            icon: "trophy.fill",
            tint: Color.appPrimary,
            primary: "Lakers @ Warriors",
            secondary: "NBA \u{00B7} Sun 1:30 PM",
            onTap: {}
        )
        SearchResultRow(
            icon: "brain.head.profile",
            tint: Color.appAccentPurple,
            primary: "Sharp Sam",
            secondary: "Public agent \u{00B7} 58% W",
            trailingDetail: "+12.4u",
            onTap: {}
        )
        SearchResultRow(
            icon: "bell.badge.fill",
            tint: Color.appAccentAmber,
            primary: "Fade Lakers",
            secondary: "NBA \u{00B7} Spread \u{00B7} 78% model",
            onTap: {}
        )
        SearchResultRow(
            icon: "sportscourt.fill",
            tint: Color.appAccentBlue,
            primary: "LAL 88 vs GSW 82",
            secondary: "NBA \u{00B7} Q4 4:21",
            trailingDetail: "LIVE",
            onTap: {}
        )
    }
    .listStyle(.insetGrouped)
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/search/SearchResultRow.kt
package com.wagerproof.app.features.search

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.core.design.icons.AppIcon
import com.wagerproof.core.design.tokens.AppColors

/**
 * Uniform list row for every Search result section — one shape for games,
 * agents, outliers and props so the result list reads as a single surface;
 * only the leading-icon tint and the secondary label change per kind. Port of
 * iOS `Features/Search/Components/SearchResultRow.swift`.
 *
 * The chevron is drawn explicitly (not a nav accessory) because result taps
 * fire imperative handoffs — switch tab + open a sheet — rather than pushing a
 * child onto a local stack.
 */
@Composable
fun SearchResultRow(
    icon: ImageVector,
    tint: androidx.compose.ui.graphics.Color,
    primary: String,
    onTap: () -> Unit,
    modifier: Modifier = Modifier,
    secondary: String? = null,
    trailingDetail: String? = null,
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clickable(onClick = onTap)
            .padding(vertical = 8.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        IconBadge(icon = icon, tint = tint)
        Spacer(Modifier.width(12.dp))
        Column(Modifier.weight(1f)) {
            Text(
                text = primary,
                color = AppColors.appTextPrimary,
                fontSize = 16.sp,
                fontWeight = FontWeight.SemiBold,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
            if (!secondary.isNullOrEmpty()) {
                Text(
                    text = secondary,
                    color = AppColors.appTextSecondary,
                    fontSize = 13.sp,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
            }
        }
        Spacer(Modifier.width(8.dp))
        if (!trailingDetail.isNullOrEmpty()) {
            Text(
                text = trailingDetail,
                color = AppColors.appTextSecondary,
                fontSize = 13.sp,
                fontWeight = FontWeight.Medium,
                fontFamily = FontFamily.Monospace,
                maxLines = 1,
            )
            Spacer(Modifier.width(8.dp))
        }
        Icon(
            imageVector = AppIcon.CHEVRON_RIGHT.imageVector,
            contentDescription = null,
            tint = AppColors.appTextMuted,
            modifier = Modifier.size(13.dp),
        )
    }
}

/** 36pt tinted rounded-square icon container (18pt symbol) — Settings-row shape. */
@Composable
private fun IconBadge(icon: ImageVector, tint: androidx.compose.ui.graphics.Color) {
    Box(
        modifier = Modifier
            .size(36.dp)
            .clip(RoundedCornerShape(8.dp))
            .background(tint.copy(alpha = 0.16f)),
        contentAlignment = Alignment.Center,
    ) {
        Icon(
            imageVector = icon,
            contentDescription = null,
            tint = tint,
            modifier = Modifier.size(16.dp),
        )
    }
}
